File: bench/langs/ledger.fs
// Keeps the running balance for one account.
module Ledger

let limit = 5000

let refuse owner =
    warnOwner owner
    0

let commitEntry owner amount =
    writeEntry owner amount
    amount

// Bills the account and returns what is left.
let charge owner amount =
    if amount > limit then
        refuse owner
    else
        commitEntry owner amount

// F# implements an interface with `override`, and a member is named after the
// receiver rather than before it. The fixture wrote only `let`, so the corpus
// is what found this: a file of interface implementations measured 42% on
// `<module>` with every member invisible.
type Ledger() =
    member this.Charge(owner, amount) =
        charge owner amount

    override this.ToString() =
        refuse 0

File: bench/langs/ledger.lua
-- Keeps the running balance for one account.

local LIMIT = 5000

local function refuse(owner, amount)
    warn_owner(owner)
    return 0
end

local function commit_entry(owner, amount)
    write_entry(owner, amount)
    return amount
end

-- Bills the account and returns what is left.
function charge(owner, amount)
    if amount > LIMIT then
        return refuse(owner, amount)
    end
    return commit_entry(owner, amount)
end
